src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_evp_pkey_rsa.c

```diff
@@ -3,6 +3,31 @@
 
 #include "pal_evp_pkey_rsa.h"
 
+EVP_PKEY* CryptoNative_RsaGenerateKey(int keySize)
+{
+    EVP_PKEY_CTX* ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_RSA, NULL);
+
+    if (ctx == NULL)
+    {
+        return NULL;
+    }
+
+    EVP_PKEY* pkey = NULL;
+    int success = 1;
```

Comment:
This code pattern feels weird to me. Seems like it's too tempting to refactor this to `success &= ...` and break the short-circuiting logic.
Isn't `if (call1() == 1 && call2() == 1 && call3() == 1)` a more established pattern?

Reply:
Yeah, I think I was sort of playing with the syntax a bit looking for "legible and compact", but didn't do it anywhere else in the EVP_PKEY experimentation.

I'll flip it around to look more uniform.